src/Adapter/Product/CommandHandler/UpdateProductPricesHandler.php

```diff
@@ -68,7 +69,26 @@ public function __construct(
      */
     public function handle(UpdateProductPricesCommand $command): void
     {
-        $product = $this->productRepository->get($command->getProductId());
+        // @todo: this part will very likely be very similar in other multistore handlers, maybe we can share this via an abstract class later
+        if (null === $command->getShopConstraint()) {
+            $this->updatePriceByShop($command);
+        } elseif ($command->getShopConstraint()->forAllShops()) {
+            $shops = $this->productRepository->getAssociatedShopIds($command->getProductId());
+            foreach ($shops as $shopId) {
+                $this->updatePriceByShop($command, $shopId);
+            }
```

Comment:
Hum, this looks like a future performance issue. The way I see it, an update for a single shop is an update performed with a constraint (eg. `WHERE shop_id = %`) and an update for multiple shops is an update without constraint (eg. `WHERE 1`). Therefore the shop constraint should naturally go all the way to `$this->productRepository->partialUpdate()` so that the repository can perform an optimized batch update query instead of a bunch of single updates.